pycbc/results/pygrb_plotting_utils.py

```diff
@@ -24,447 +24,26 @@
 """
 
 import sys
-import os
-import logging
-import argparse
 import copy
 import numpy
-# TODO: imports to fix/remove
-try:
-    from ligo import segments
-    from ligo.lw import utils, lsctables
-except ImportError:
-    pass
-try:
-    from pylal import MultiInspiralUtils
-    from pylal.coh_PTF_pyutils import get_bestnr, get_det_response
-    from pylal.coh_PTF_pyutils import readSegFiles
-    from pylal.dq import dqSegmentUtils
-except ImportError:
-    pass
+from pycbc.results import save_fig_with_metadata
 # Only if a backend is not already set ... This should really *not* be done
 # here, but in the executables you should set matplotlib.use()
 # This matches the check that matplotlib does internally, but this *may* be
-# version dependenant. If this is a problem then remove this and control from
+# version dependent. If this is a problem then remove this and control from
 # the executables directly.
-import matplotlib
 if 'matplotlib.backends' not in sys.modules:  # nopep8
+    import matplotlib
     matplotlib.use('agg')
```

Comment:
Please don't set this in any library code (as it can cause issues for people running Jupyter). Please put this in any executable that needs it instead.